packages/block-library/src/latest-posts/edit.js

```diff
@@ -363,20 +363,36 @@ class LatestPostsEdit extends Component {
 							[ `align${ featuredImageAlign }` ]: !! featuredImageAlign,
 						} );
 
-						const postExcerpt =
+						const needsReadMore =
 							excerptLength <
 								excerpt.trim().split( ' ' ).length &&
-							post.excerpt.raw === ''
-								? excerpt
+							post.excerpt.raw === '';
+
+						const getReadMoreLink = ( excerptText, link ) => {
```

Comment:
This function is being reinstantiated on every iteration of `displayPosts.map`. One fix would be to push it out, but in this case I would probably just inline its definition, i.e.

```js
const needsReadMore = …;
const postExcerpt = needsReadMore
    ? (
        <>
            { excerptText.trim().split( ' ', excerptLength ).join( ' ' ) }
            …
        </>
    ) : excerpt;
```